src/main/java/org/folio/services/locationunit/CampusService.java

```diff
@@ -0,0 +1,113 @@
+import org.folio.rest.jaxrs.model.Loccamps;
+import org.folio.rest.jaxrs.resource.LocationUnits.DeleteLocationUnitsCampusesByIdResponse;
+import org.folio.rest.jaxrs.resource.LocationUnits.DeleteLocationUnitsCampusesResponse;
+import org.folio.rest.jaxrs.resource.LocationUnits.GetLocationUnitsCampusesByIdResponse;
+import org.folio.rest.jaxrs.resource.LocationUnits.GetLocationUnitsCampusesResponse;
+import org.folio.rest.jaxrs.resource.LocationUnits.PostLocationUnitsCampusesResponse;
+import org.folio.rest.jaxrs.resource.LocationUnits.PutLocationUnitsCampusesByIdResponse;
+import org.folio.rest.persist.PgUtil;
+import org.folio.services.domainevent.CampusDomainEventPublisher;
+
+public class CampusService {
+
+  private static final String MSG_ID_NOT_MATCH = "Illegal operation: Campus ID cannot be changed";
+
+  private final Context context;
+  private final Map<String, String> okapiHeaders;
+  private final CampusRepository repository;
+  private final CampusDomainEventPublisher domainEventService;
+
+  public CampusService(Context context, Map<String, String> okapiHeaders) {
+    this.context = context;
+    this.okapiHeaders = okapiHeaders;
+    this.repository = new CampusRepository(context, okapiHeaders);
+    this.domainEventService =
+      new CampusDomainEventPublisher(context, okapiHeaders);
+  }
+
+  public Future<Response> getByQuery(String cql, int offset, int limit) {
+    return PgUtil.get(CAMPUS_TABLE, Loccamp.class, Loccamps.class,
+      cql, offset, limit, okapiHeaders, context,
+      GetLocationUnitsCampusesResponse.class);
+  }
+
+  public Future<Response> getById(String id) {
+    return PgUtil.getById(CAMPUS_TABLE, Loccamp.class, id, okapiHeaders, context,
+      GetLocationUnitsCampusesByIdResponse.class);
+  }
+
+  public Future<Response> create(Loccamp loccamp) {
+    return PgUtil.post(CAMPUS_TABLE, loccamp, okapiHeaders, context,
+        PostLocationUnitsCampusesResponse.class)
+      .onSuccess(response ->
+        PostLocationUnitsCampusesResponse
+          .respond201WithApplicationJson(response,
+            PostLocationUnitsCampusesResponse.headersFor201()))
+      .otherwise(throwable ->
+        PostLocationUnitsCampusesResponse.respond500WithTextPlain(
+          throwable.getMessage()));
+  }
+
+  public Future<Response> update(String id, Loccamp loccamp) {
+    if (loccamp.getId() != null && !loccamp.getId().equals(id)) {
+      return succeededFuture(
+        PutLocationUnitsCampusesByIdResponse.respond400WithTextPlain(
+          createValidationErrorMessage(
+            "loccamp", loccamp.getId(), MSG_ID_NOT_MATCH)));
+    }
+    if (loccamp.getId() == null) {
+      loccamp.setId(id);
+    }
+
+    return repository.getById(id)
+      .compose(oldLoccamp ->
+        PgUtil.put(CAMPUS_TABLE, loccamp, id, okapiHeaders, context,
+            PutLocationUnitsCampusesByIdResponse.class)
+          .onSuccess(domainEventService.publishUpdated(oldLoccamp))
+          .otherwise(throwable ->
```

Comment:
do we need this otherwise?
For example in LibraryService we don't have it: https://github.com/folio-org/mod-inventory-storage/pull/1029/files#diff-a504c82181f298bf1db3c82e991febfb96b01ed3ed5d5295ea46050e1a2b2fcfR72

Reply:
Removed it